pennylane/devices/qubit/preprocess.py

```diff
@@ -173,27 +175,36 @@ def validate_and_expand_adjoint(
 
     # Check validity of measurements
     measurements = []
-    for m in circuit.measurements:
+    for m in tape.measurements:
         if not isinstance(m, ExpectationMP):
-            return DeviceError(
+            raise DeviceError(
```

Comment:
it was originally this way, but @albi3ro [mentioned](https://github.com/PennyLaneAI/pennylane/pull/4037#discussion_r1181734295) that she would prefer to postpone this handling. I suppose things have gotten more complicated since then, but I figured I'd loop her in and make sure this is fine before moving forward.

Reply:
I'm fine with this. makes more sense now anyway